lib/msal-react/src/MsalAuthentication.tsx

```diff
@@ -1,26 +1,25 @@
-import { AuthenticationResult } from "@azure/msal-browser";
 import React, { useState, useCallback, useEffect, useMemo } from "react";
+import { PopupRequest, RedirectRequest, SsoSilentRequest } from "@azure/msal-browser";
 
 import { IMsalContext } from "./MsalContext";
 import { useMsal } from "./MsalProvider";
-import { getChildrenOrFunction, defaultLoginHandler } from "./utilities";
+import { getChildrenOrFunction } from "./utilities";
 import { useIsAuthenticated } from "./useIsAuthenticated";
+import { InteractionType } from "Constants";
 
 export interface IMsalAuthenticationProps {
     username?: string;
-    loginHandler?: (context: IMsalContext) => Promise<AuthenticationResult>;
+    interactionType?: string;
+    authenticationRequest?: PopupRequest|RedirectRequest|SsoSilentRequest
 }
 
 type MsalAuthenticationResult = {
     error: Error | null;
     msal: IMsalContext;
 };
 
-// TODO: Add optional argument for the `request` object?
-export function useMsalAuthentication(
-    args: IMsalAuthenticationProps = {}
-): MsalAuthenticationResult {
-    const { username, loginHandler = defaultLoginHandler } = args;
+export function useMsalAuthentication(args: IMsalAuthenticationProps = {}): MsalAuthenticationResult {
+    const { username, interactionType = InteractionType.POPUP, authenticationRequest } = args;
```

Comment:
For both Angular and React, we may want to consider making interactionType required.

Reply:
I think that's reasonable. It would make it more clear to the user what's happening instead of silently defaulting to popups

Reply:
hi guys, a question about the use of `enum` instead of a string literal type for `InteractionType`?

```
interactionType = 'popup'
```
seems a lot cleaner to me, and also benefits from enhanced vscode intellisense as well as not having to import the `InteractionType` due to the string literal type being inferred by TypeScript.

With your blessing, happy to raise a PR for any of these changes.

Reply:
I would argue that enums provide better intellisense, given they provide a clear definition of the supported values. We also want to avoid magic strings whenever possible, especially given this enum is used internally by the library. And this enum does map to string values: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/src/utils/BrowserConstants.ts#L74-L78

Reply:
Thanks for the explanation. Completely understand about the decision particularly around the `magic string` point 😀